ledger/internal/eval.go

```diff
@@ -1267,6 +1270,8 @@ func (eval *BlockEvaluator) endOfBlock() error {
 		return err
 	}
 
+	eval.state.mods.OptimizeAllocatedMemory(eval.maxAcctLookback, eval.proto.MaxTxnLife)
```

Comment:
what if we call the optimization in `ledger.AddBlock` ?

Reply:
yes that makes sense